File: 448. Find All Numbers Disappeared in an Array/448.cpp
class Solution {
public:
    vector<int> findDisappearedNumbers(vector<int>& nums) {
        vector<int> ans;
        for (int i=0; i<nums.size(); ++i)
            nums[(nums[i]&INT_MAX)-1]|= ~INT_MAX;
        for (int i=0; i<nums.size(); ++i)
            if (!(nums[i]>>31&1)) ans.push_back(i+1);
        return ans;
    }
};

File: 448. Find All Numbers Disappeared in an Array/448.py
class Solution(object):

    def findDisappearedNumbers(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        mask = 1 << 31
        for i in nums:
            nums[(i & ~mask) - 1] |= mask
        return [i + 1 for i in range(len(nums)) if not(nums[i] >> 31 & 1)]
